sql/2023/06/ttfb-by-language.sql

```diff
@@ -0,0 +1,70 @@
+# HTTP Archive query to get TTFB information of WordPress sites by language.
+#
+# WPP Research, Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See https://github.com/GoogleChromeLabs/wpp-research/pull/54
+
+SELECT
+  client,
+  lang,
+  COUNT(DISTINCT origin) AS n,
+  SUM(
+    IF
+      (ttfb.start < 200, ttfb.density, 0)) / SUM(ttfb.density) AS fast,
+  SUM(
+    IF
+      (ttfb.start >= 200
+         AND ttfb.start < 1000, ttfb.density, 0)) / SUM(ttfb.density) AS avg,
+  SUM(
+    IF
+      (ttfb.start >= 1000, ttfb.density, 0)) / SUM(ttfb.density) AS slow
+FROM
+  `chrome-ux-report.all.202304`,
```

Comment:
is 202304 the right table? the other tables are 05-2023.

Reply:
There is no May table in for CrUX yet.

The HTTP Archive table is for 2023-05-01, which I assumed means it's the analyzed data from April as well.

Reply:
@swissspidy The HTTP Archive tables always use the 1st day of the month, but that doesn't mean anything really, technically speaking those could just be called "2023-05". The pipelines are run later than that, so they contain data from _some point_ in May.

CrUX data however is only always released the month after (usually 2nd Tuesday of each month), for example May data was published on June 13 (see https://developer.chrome.com/docs/crux/release-notes/).

I'd argue for an analysis like this it doesn't matter all that much whether we look at the latest month or some time 1-2 months back, so if the latest data is not available, we can simply pick a previous month. But at this point you should be able to consistently use `2023-05-01` for both HTTP Archive and CrUX.

Reply:
Last I checked (a month ago) that table wasn‘t available yet, but I‘ll check again